x-pack/plugins/canvas/server/lib/build_es_request.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { buildBoolArray } from './build_bool_array';
+import {
+  CanvasQueryFilter,
+  ElasticsarchTermFilter,
+  ElasticsearchLuceneQueryStringFilter,
+  ElasticsearchTimeFilter,
+} from './filters';
+
+type ElasticsearchFilter =
```

Comment:
This is `copypasta`.  I would add this type to a `filters.ts` and import from there.